src/System.Net.Mail/tests/Functional/CredentialTest.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Mail.Functional.Tests
+{
+    public class CredentialTest
```

Comment:
If this is testing SmtpClient the test class should be named appropriately, e.g. `SmtpClientTests` or `SmtpClientCredentialsTests`. @davidsh can probably provide better guidance on naming conventions.

Reply:
>@davidsh can probably provide better guidance on naming conventions.

I would name the class (and the filename) as `SmtpClientCredentialsTest`.

We usually always suffix the name with 'Test' not 'Tests' based on our following the guidance from the 'Art of Unit Testing" book (however, we have some deviations from this).

Since this test class and methods are testing both the 'UseDefaultCredentials' and 'Credentials' properties of the SmtpClient class, using "SmtpClientCredentialsTest" seems the most appropriate name.